src/identityApiClient.ts

```diff
@@ -99,55 +114,76 @@ export default function IdentityAPIClient(
         try {
             const response: Response = await uploader.upload(uploadPayload);
 
-            let message: string;
             let aliasResponseBody: IAliasResponseBody;
-
-            // FetchUploader returns the response as a JSON object that we have to await
-            if (response.json) {
-                // HTTP responses of 202, 200, and 403 do not have a response.  response.json will always exist on a fetch, but can only be await-ed when the response is not empty, otherwise it will throw an error.
-                try {
-                    aliasResponseBody = await response.json();
-                } catch (e) {
-                    verbose('The request has no response body');
-                }
-            } else {
-                // https://go.mparticle.com/work/SQDSDKS-6568
-                // XHRUploader returns the response as a string that we need to parse
-                const xhrResponse = (response as unknown) as XMLHttpRequest;
-
-                aliasResponseBody = xhrResponse.responseText
-                    ? JSON.parse(xhrResponse.responseText)
-                    : '';
-            }
-
+            let message: string;
             let errorMessage: string;
 
             switch (response.status) {
-                case HTTP_OK:
                 case HTTP_ACCEPTED:
-                    // https://go.mparticle.com/work/SQDSDKS-6670
-                    message =
-                        'Successfully sent forwarding stats to mParticle Servers';
-                    break;
-                default:
-                    // 400 has an error message, but 403 doesn't
-                    if (aliasResponseBody?.message) {
-                        errorMessage = aliasResponseBody.message;
+                case HTTP_OK:
+
+                // 400 error will has a body and will go through the happy path to report the error
+                case HTTP_BAD_REQUEST:
+
+                    // FetchUploader returns the response as a JSON object that we have to await
+                    if (response.json) {
+                        // HTTP responses of 202, 200, and 403 do not have a response.
+                        // response.json will always exist on a fetch, but can only be
+                        // await-ed when the response is not empty, otherwise it will
+                        // throw an error.
+                        try {
+                            aliasResponseBody = await response.json();
+                        } catch (e) {
+                            verbose('The request has no response body');
+                        }
```

Comment:
Previously, we had this message because we weren't doing it inside of a switch statement, so it technically allowed for all other http status codes to be .  However, This is a bit confusing to me.  We say 202, 200, and 403 won't have a body, and so this switch statement will _always_ throw an error and go to the catch statement, right?

It also seems like `aliasResponseBody` is only ever used for grabbing the errors, and not in a successful alias request.  I feel like this function is still confusing. I think it should handle 200 and 202 separately from errors.

Reply:
Doing it in this switch case requires us to know that the only responses coming back are 202, 200, and 403. otherwise, it goes to the default and throws an error.  ultimately everything just calls `invokeAliasCallback` which is the main concern, but I have a small worry that there may be an error for a 20X that we aren't catching, and then we are telling the customer via a thrown Error that there's a 204, or 203.  not sure if these are realistic use cases, but I think that's what the code was doing in the past via verbose, and it not necessarily being an error.  Perhaps we should sync w Napo to get an idea of _every_ status code that can come back